### PR TITLE
Add uid alias to data concepts that pulls the citrine id.

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
This is a convenience method that makes the data concepts resources
look more like the other resources.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

**Review by bfolie — APPROVED** [review]
nice

File: tests/resources/test_material_run.py [hunk 1]
```diff
@@ -80,6 +80,8 @@ def test_nomutate_gemd(collection, session):
     # Then
     assert before == after
```
Comment (kroenlein) [inline tests/resources/test_material_run.py:81]: I'm actually a little concerned about what `==` means in this context.  Is this all the behavior we actually wanted?

**Review by kroenlein — APPROVED** [review]
I approve of the immediate change, but I'm not sure I agree with some of the existing mutation logic.  I thought `==` objects were `id() == id()`, which should not work here.

File: tests/resources/test_material_run.py (hunk 1 above)
Comment (maxhutch) [reply tests/resources/test_material_run.py:81]: I believe that it inherits from [DictSerializable](https://github.com/CitrineInformatics/gemd-python/blob/main/gemd/entity/dict_serializable.py#L138).  Value based equality seems appropriate here.

**Pull request merged**